sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/ColumnPruningSuite.scala

```diff
@@ -360,5 +360,34 @@ class ColumnPruningSuite extends PlanTest {
     comparePlans(optimized2, expected2.analyze)
   }
 
+  test("SPARK-21707 the condition of filter is not deterministic that split to two project ") {
```

Comment:
Actually I don't get what the test title tries to say. Can you try to rephrase it?